packages/fast-components-react-base/src/viewport-positioner/README.md

```diff
@@ -0,0 +1,53 @@
+
+### Usage
+
+The *viewport positioner* and the anchor element it is attached to must both be children of the specified viewport element. There should also be no other scrollable elements in the DOM hierarchy between these elements and the viewport.  The *viewport positioner* element will have its position attribute programmatically set to "relative" and the component also writes in values for the top, right, bottom, left, transform-origin and tranform attributes. 
+
+If no "anchor" is specified the component will be disabled and not attempt to adjust layout. If no viewport is specified the component will assume the viewport is the root document element of the page.
+
+Authors can specify a positioning mode for each axis or leave it "uncontrolled" and use other means (ie. their own css) to manage the positioner's layout on that axis.  The "flipOutward" positioning mode ensures that the positioner does not overlap with the anchor on that axis and butts up against the outer edges of the anchor element.  With the "flipInward" positioning mode set the positioner will overlap with the anchor on the axis specified but will only extend beyond the anchor on the side with the most available space.
+
+FlipOutward:
+
+Anchor
+       |Positioner
+
+ or
+
+             Anchor
+ Positioner|
+
+ FlipInward:
+
+ Anchor
+ |Positioner
+
+ or 
+
+      Anchor
+ Positioner|
+
+ The positioning mode chosen determines the positions that can be chosen.  The "flipOutward" mode allows the "left" and "right" positions only on the horizontal axis, while the "flipInward" mode allows the "centerLeft" and "centerRight" positions. The vertical axis behaves similarly though the positions are named differently.  
+ 
+ If a default position is specified for a particular axis and there is enough space that position will be set.  Whether there is enough space is determined by comparing the threshold size for the axis (or the height/width of the positioner if no threshold specified) to the available space for that position. If there is not enough space, or if no default position is specified, the positioner chooses the position with the most available space.
+
+The chosen position determines which side of the positioner is fixed to the anchor.  For example if the chosen position for the horizontal axis is "left" the positioner will have its "right" property set to a value corresponding to the left edge of the anchor and the "left" propertly will be unset.
+
+The *viewport positioner* adds css classes to itself based on the chosen positions (ie. "viewportPositioner__left", "viewportPositioner__right", etc...) to enable authors to style the positioner and its contents based on relative position.
+
+The "AlwaysInView" props for each axis allows authors to specify whether the positioner should remain on screen when the anchor element is scrolled off.  The *viewport positioner* accomplishes this by setting the translate transform and translate origin properties of the component.
```

Comment:
I would enumerate these props to be more clear (i.e. horizontalAlwaysInView and verticalAlwaysInView

Reply:
yep